test/Libraries/DynamoMSOfficeTests/ExcelTests.cs

```diff
@@ -335,57 +313,73 @@ public void CanReadAndWriteExcel()
 
             Assert.IsTrue(File.Exists(filePath));
 
-            Assert.IsTrue(writeNode.CachedValue.IsCollection);
-            var list1 = writeNode.CachedValue.GetElements();
-
-            Assert.IsTrue(watch.CachedValue.IsCollection);
-            var list2 = watch.CachedValue.GetElements();
+            var data = new object[] { new object[] { 1 }, new object[] { "word" }, new object[] { 2 }, new object[] { 3 }, new object[] { "palabra" } };
 
-            Assert.AreEqual(5, list1.Count());
-            Assert.AreEqual(5, list2.Count());
+            AssertPreviewValue(watch.GUID.ToString(), data);
 
-            // single column - 1, "word", 2, 3, "palabra"
-            Assert.IsTrue(list2[0].IsCollection);
-            var rowList = list2[0].GetElements();
-            Assert.AreEqual(1, rowList[0].Data);
-
-            Assert.IsTrue(list2[1].IsCollection);
-            rowList = list2[1].GetElements();
-            Assert.AreEqual("word", rowList[0].Data);
-
-            Assert.IsTrue(list2[2].IsCollection);
-            rowList = list2[2].GetElements();
-            Assert.AreEqual(2, rowList[0].Data);
+            
+        }
+        [Test]
+        public void ReadChangeFilename()
+        {
 
-            Assert.IsTrue(list2[3].IsCollection);
-            rowList = list2[3].GetElements();
-            Assert.AreEqual(3, rowList[0].Data);
+            // 1.  Read from an Existing excel file and Run
+            // 2.  Modify the file name in file path to an existing file
+            //3.  Rerun the file  
+            string openPath = Path.Combine(TestDirectory, @"core\excel\ChangeFilename.dyn");
+            ViewModel.OpenCommand.Execute(openPath);
 
-            Assert.IsTrue(list2[4].IsCollection);
-            rowList = list2[4].GetElements();
-            Assert.AreEqual("palabra", rowList[0].Data);
+            var filename = ViewModel.Model.CurrentWorkspace.FirstNodeFromWorkspace<DSCore.File.Filename>();
 
+            // remap the filename as Excel requires an absolute path
+            filename.Value = filename.Value.Replace(@"..\..\..\test", TestDirectory);
+   
+            ViewModel.HomeSpace.Run();
+            
+            var watch = ViewModel.Model.CurrentWorkspace.GetDSFunctionNodeFromWorkspace("Excel.ReadFromFile");
+            var watch2 = ViewModel.Model.CurrentWorkspace.NodeFromWorkspace("b9b04f1f-9069-4eaf-a31c-eee7428aacab");
```

Comment:
The guid here represent FileName node, so better use fileNameNode instead of watch2 and likewise watch3 should be renamed as fileObjectNode, so when you later on connect these nodes, it will make sense.